yt/data_objects/time_series.py

```diff
@@ -435,6 +440,179 @@ def particle_trajectories(
             self, indices, fields=fields, suppress_logging=suppress_logging, ptype=ptype
         )
 
+    def _get_by_attribute(
+        self,
+        attribute: str,
+        value: Union[unyt_quantity, tuple[float, str]],
+        tolerance: Union[None, unyt_quantity, tuple[float, str]] = None,
+        side: Union[
+            Literal["nearest"], Literal["smaller"], Literal["larger"]
+        ] = "nearest",
+    ) -> "Dataset":
+        r"""
+        Get a dataset at or near to a given value.
+
+        Parameters
+        ----------
+        key : str
+            The key by which to retrieve an output, usually 'current_time' or
+            'current_redshift'. The key must be an attribute of the dataset
+            and monotonic.
+        value : unyt_array or (value, unit)
+            The value to search for.
+        tolerance : unyt_array or (value, unit)
+            If not None, do not return a dataset unless the value is
+            within the tolerance value. If None, simply return the
+            nearest dataset.
+            Default: None.
+        side : str
+            The side of the value to return. Can be 'nearest', 'smaller' or 'larger'.
+            Default: 'nearest'.
+        """
+
+        if side not in ("nearest", "smaller", "larger"):
+            raise ValueError(
+                f"side must be 'nearest', 'smaller' or 'larger', got {side}"
+            )
+
+        # Use a binary search to find the closest value
+        iL = 0
+        iH = len(self._pre_outputs) - 1
```

Comment:
I don't understand what `H` stands for here. I would expect `iL` and `iR` (left/right) be used instead, can you explain ?